test-suite/e2e/test/delegateUserDecrypt/delegateUserDecryption.ts

```diff
@@ -0,0 +1,51 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+
+import { createInstances } from '../instance';
+import { getSigners, initSigners } from '../signers';
+import { userDecryptSingleHandle } from '../utils';
+
+describe('Delegate user decryption', function () {
+  before(async function () {
+    await initSigners(2);
+    this.signers = await getSigners();
+    this.instances = await createInstances(this.signers);
+    const contractFactory = await ethers.getContractFactory('DelegateUserDecryptDelegator');
+
+    this.contract = await contractFactory.connect(this.signers.alice).deploy();
+    await this.contract.waitForDeployment();
+    this.contractAddress = await this.contract.getAddress();
+
+    const contractFactoryDelegate = await ethers.getContractFactory('DelegateUserDecryptDelegate');
+
+    this.contractDelegate = await contractFactoryDelegate.connect(this.signers.alice).deploy();
+    await this.contractDelegate.waitForDeployment();
+    this.contractDelegateAddress = await this.contractDelegate.getAddress();
+
+    this.instances = await createInstances(this.signers);
+  });
+
+  const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+  it('test delegate user decrypt ebool', async function () {
```

Comment:
Not sure, but it seems that this test case is testing the delegate and revoke use cases, not any decryption, right? In such a case, maybe better to specify this, e.g., `test delegation and revocation of user decryption`